crates/viewer/re_ui/src/notifications.rs

```diff
@@ -0,0 +1,393 @@
+                                }
+                            });
+                        });
+                        egui::ScrollArea::vertical()
+                            .min_scrolled_height(panel_max_height)
+                            .show(ui, notification_list);
+
+                        if !notifications.is_empty() {
+                            ui.horizontal_top(|ui| {
+                                if ui.button("Clear all").clicked() {
+                                    clear = true;
+                                };
+                            });
+                        }
+                    });
+            });
+
+        if clear {
+            notifications.clear();
+        } else if let Some(to_dismiss) = to_dismiss {
+            notifications.remove(to_dismiss);
+        }
+    }
+}
+
+fn base_ttl() -> Duration {
+    Duration::from_secs_f64(4.0)
+}
+
+struct Toasts {
+    id: egui::Id,
+}
+
+impl Default for Toasts {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Toasts {
+    fn new() -> Self {
+        Self {
+            id: egui::Id::new("__toasts"),
+        }
+    }
+
+    /// Shows and updates all toasts
+    fn show(&self, egui_ctx: &egui::Context, notifications: &mut [Notification]) {
+        let dt = Duration::from_secs_f32(egui_ctx.input(|i| i.unstable_dt));
+        let mut offset = egui::vec2(-8.0, 32.0);
+
+        for (i, notification) in notifications
+            .iter_mut()
+            .enumerate()
+            .filter(|(_, n)| n.ttl > Duration::ZERO)
+        {
+            let response = egui::Area::new(self.id.with(i))
+                .anchor(egui::Align2::RIGHT_TOP, offset)
+                .order(egui::Order::Foreground)
+                .interactable(true)
+                .movable(false)
+                .show(egui_ctx, |ui| {
+                    show_notification(ui, notification, DisplayMode::Toast, || {})
+                })
+                .response;
+
+            if !response.hovered() {
+                if notification.ttl < dt {
+                    notification.ttl = Duration::ZERO;
+                } else {
+                    notification.ttl -= dt;
+                }
+            }
+
+            let response = response.on_hover_text("Click to close and copy contents");
+
+            if response.clicked() {
+                egui_ctx.output_mut(|o| o.copied_text = notification.text.clone());
+                notification.ttl = Duration::ZERO;
+            }
+
+            offset.y += response.rect.height() + 8.0;
+        }
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum DisplayMode {
+    Panel,
+    Toast,
+}
+
+fn show_notification(
+    ui: &mut egui::Ui,
+    notification: &Notification,
+    mode: DisplayMode,
+    mut on_dismiss: impl FnMut(),
+) -> egui::Response {
+    egui::Frame::window(ui.style())
+        .rounding(4.0)
+        .inner_margin(10.0)
+        .fill(hex_color!("#1c2123"))
```

Comment:
`crates/viewer/re_ui/src/color_table.rs`

maybe we should add a lint for this :)